apiserver/facades/agent/secretsmanager/secrets.go

```diff
@@ -412,6 +420,51 @@ func (s *SecretsManagerAPI) WatchSecretsChanges(args params.Entities) (params.St
 	return results, nil
 }
 
+// WatchObsoleteRevisions sets up a watcher to notify of changes to obsolete secret revisions for the specified secrets.
+func (s *SecretsManagerAPI) WatchObsoleteRevisions(args params.Entities) (params.StringsWatchResults, error) {
+	results := params.StringsWatchResults{
+		Results: make([]params.StringsWatchResult, len(args.Entities)),
+	}
+	one := func(arg params.Entity) (string, []string, error) {
+		tag, err := names.ParseTag(arg.Tag)
+		if err != nil {
+			return "", nil, errors.Trace(err)
+		}
+		if !s.isSameApplication(tag) {
+			return "", nil, apiservererrors.ErrPerm
+		}
+
+		// Only unit leaders can watch secrets owned by the app.
+		appName := authTagApp(s.authTag)
+		if tag.Kind() == names.ApplicationTagKind {
+			token := s.leadershipChecker.LeadershipCheck(appName, s.authTag.Id())
+			if err := token.Check(0, nil); err != nil {
+				return "", nil, errors.Trace(err)
+			}
+		}
+
+		w := s.secretsConsumer.WatchObsoleteRevisions(arg.Tag)
+		if obsolete, ok := <-w.Changes(); ok {
+			changes := make([]string, len(obsolete))
```

Comment:
You know I love this more
`changes := append([]string(nil), obsolete...)`

Reply:
Do we even need the copy. Can't obsolete just be returned here. I can't see a reason why we need to shift it.

Reply:
probably don't need it